lib/galaxy/metadata/set_metadata.py

```diff
@@ -338,9 +346,17 @@ def set_meta(new_dataset_instance, file_dict):
                 input_ext=input_ext,
             )
             collect_dynamic_outputs(job_context, output_collections)
-        except MaxDiscoveredFilesExceededError as e:
+        except MaxDiscoveredFilesExceededError:
             final_job_state = Job.states.ERROR
-            job_messages.append(str(e))
+            job_messages.append(
+                {
+                    "type": "max_discovered_files",
+                    "desc": "Too many output files were discovered for a single job.",
```

Comment:
```suggestion
                    "desc": str(e),
```
I'm sorry, that's actually a better message since it includes the limit.